internal/providers/configdrive/configdrive.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The configdrive provider fetches a user_data from a config-2 file system.
+
+package configdrive
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/internal/log"
+	"github.com/coreos/ignition/internal/providers"
+	"github.com/coreos/ignition/internal/providers/util"
+)
+
+const (
+	initialBackoff = 100 * time.Millisecond
+	maxBackoff     = 30 * time.Second
+	configDevice   = "/dev/disk/by-label/config-2"
+	configPath     = "/openstack/latest/user_data"
+)
+
+// These constants come from <cdrom.h>.
```

Comment:
Most of this looks like a direct copy/paste from the Azure provider. Can you move the common code into a shared set of functions or a utility package?